### PR TITLE
Update datadog exporter codeowners

### PR DESCRIPTION
**Pull request by liustanley**
Add myself as a code owner

### CODE REVIEW

**Comment by djaglowski** [comment]
@evan-bradley, should this have pinged the codeowners once I added the component tag?

**Comment by djaglowski** [comment]
@KSerrania @mx-psi @gbbr @knusbaum @amenasria @dineshg13, please confirm.

**Comment by evan-bradley** [comment]
> @evan-bradley, should this have pinged the codeowners once I added the component tag?

The workflow should add code owners as reviewers to a PR once the label is added, but unfortunately GitHub doesn't allow adding reviewers who are not OpenTelemetry organization members, which is what happened here. There's an open issue to track leaving a comment to ping code owners in these cases here: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16326.

**Review by dineshg13 — APPROVED** [review]
LGTM . @liustanley is a member of our team .

**Review by mx-psi — DISMISSED** [review]
LGTM, but @liustanley is not a member of the OTel org, so we should probably figure that out first before merging this.

**Comment by mx-psi** [comment]
> The workflow should add code owners as reviewers to a PR once the label is added, but unfortunately GitHub doesn't allow adding reviewers who are not OpenTelemetry organization members, which is what happened here.

I think only two out of the six listed here are not members, did it fail because of those two?

File: .github/CODEOWNERS
Comment (mx-psi) [inline .github/CODEOWNERS:43]: We can also take this opportunity to remove @KSerrania @knusbaum and @amenasria  from here since they are no longer actively working on the exporter (will reach out for their approvals)

```suggestion
exporter/datadogexporter/                            @open-telemetry/collector-contrib-approvers @mx-psi @gbbr @dineshg13 @liustanley
```

Comment (knusbaum) [reply .github/CODEOWNERS:43]: works for me, yes.

**Review by knusbaum — APPROVED** [review]

File: .github/CODEOWNERS
Comment (amenasria) [reply .github/CODEOWNERS:43]: Hey, yes you can remove me, no prob !! 🫡

**Review by amenasria — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
> LGTM, but @liustanley is not a member of the OTel org, so we should probably figure that out first before merging this.

@mx-psi did you want to block the PR until this is resolved?

**Comment by TylerHelmuth** [comment]
We've enforced this CODEOWNER requirement before, I don't think we should bend the rules in this case. The requirement is there so that assignments can work.

**Comment by codeboten** [comment]
> We've enforced this CODEOWNER requirement before, I don't think we should bend the rules in this case. The requirement is there so that assignments can work.

Sure, to clarify I was mostly asking @mx-psi to block the PR until that was resolved :D

**Comment by TylerHelmuth** [comment]
@mx-psi and I have talked about this situation before, it's a bit of a catch-22 haha

**Comment by mx-psi** [comment]
> @mx-psi did you want to block the PR until this is resolved?

@codeboten I meant to comment instead of approving. I have talked about this with @TylerHelmuth before, I don't feel like it is a big deal to have codeowners that are not org members, but I don't think I should ignore the rules just because I don't agree with them :smile:

File: .github/CODEOWNERS
Comment (KSerrania) [reply .github/CODEOWNERS:43]: Looks good to me.

**Review by KSerrania — APPROVED** [review]

**Review by dineshg13 — APPROVED** [review]

**Comment by mx-psi** [comment]
Per open-telemetry/community/issues/1349 this can be merged after we fix the merge conflict :tada:  Welcome @liustanley :) let me know when the conflict is addressed.

**Comment by liustanley** [comment]
Added @songy23 as a codeowner as well since he's also on our team

**Review by codeboten — APPROVED** [review]

**Pull request merged**
